bofire/strategies/predictives/enting.py

```diff
@@ -0,0 +1,128 @@
+from typing import List, Tuple
+
+import numpy as np
+import pandas as pd
+import pyomo.environ as pyo
+from entmoot.models.enting import Enting
+from entmoot.optimizers.pyomo_opt import PyomoOptimizer
+from entmoot.problem_config import ProblemConfig
+from pydantic import PositiveInt
+
+import bofire.data_models.strategies.api as data_models
+from bofire.data_models.features.api import TInputTransformSpecs
+from bofire.data_models.objectives.api import MaximizeObjective
+from bofire.strategies.predictives.predictive import PredictiveStrategy
+from bofire.utils.entmoot import domain_to_problem_config
+
+
+class EntingStrategy(PredictiveStrategy):
+    """Strategy for selecting new candidates using ENTMOOT"""
+
+    def __init__(
+        self,
+        data_model: data_models.EntingStrategy,
+        **kwargs,
+    ):
+        super().__init__(data_model=data_model, **kwargs)
+        self._init_problem_config()
+        self._enting = Enting(self._problem_config, data_model.enting_params)
+        self._solver_params = data_model.solver_params
+        self._learn_from_candidates_coeff = data_model.learn_from_candidates_coeff
+
+    def _init_problem_config(self) -> None:
+        cfg = domain_to_problem_config(self.domain)
+        self._problem_config: ProblemConfig = cfg[0]
+        self._model_pyo: pyo.ConcreteModel = cfg[1]
+
+    @property
+    def input_preprocessing_specs(self) -> TInputTransformSpecs:
```

Comment:
If not needed, can you just remove this method, or is it enforced by the base class?

Reply:
Frustratingly, it's it is needed. The parent method is an `abstractmethod`. Also, if the function returns `None`, then the line below throws an error, since you're trying `None.items()`.

https://github.com/experimental-design/bofire/blob/4790d637016f4e482ae8bafa6cb711b66e26b99e/bofire/data_models/domain/features.py#L502